helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -591,8 +591,11 @@ public void addCurrentStateChangeListener(org.apache.helix.CurrentStateChangeLis
   @Override
   public void addCustomizedStateRootChangeListener(CustomizedStateRootChangeListener listener,
       String instanceName) throws Exception {
-    addListener(listener, new Builder(_clusterName).customizedStatesRoot(instanceName),
-        ChangeType.CUSTOMIZED_STATE_ROOT, new EventType[]{EventType.NodeChildrenChanged});
+    PropertyKey propertyKey = _dataAccessor.keyBuilder().customizedStatesRoot(instanceName);
+    if (_zkclient.exists(propertyKey.getPath())) {
```

Comment:
I am concerned about the performance. This check adds more time for adding listeners because of network latency. I wonder if there is a better way so this check could be avoided.

Reply:
If we don't check path, the only way I can think of is to check participant side version.  Doesn't sounds a clean way though.

Reply:
Maybe we could just change the the log in callback handler not to log the exception stack trace.

Reply:
That would impact all the exceptions right? Then it's not good for debugging purpose.

Reply:
It is just a warn log. I think it is OK not to print the stack trace. And if the stack trace is necessary, we could keep the stack trace for the other paths, but for this customized state path, just log the message.

Reply:
exists() check is not a heavyweight ZK operation and considering we only call this when there's a LiveInstance change (or periodic rebalance), I don't think performance is a concern here.

Reply:
Dummy question, why participant needs to listen on customized view changes? Only controller needs to listen on the change, not?

Reply:
Participant does not need to listen on customized view change. The backward compatibility issue happens when we have a new version controller, while the client is on an old version participant. The new path INSTANCES/CUSTOMIZED_STATES does not exist, but controller will try to add listener to that path during live instance change. Then zkclient will have a no node exception. It's fine, just a warn, but kind of pollute client's log and make them confused.

Reply:
Shall we have some logs to indicate it is not able to subscribe? Maybe warning or something?

Reply:
I updated this rb with a solution to avoid stack trace in our customers who have not updated their helix-lib version. As discussed before, this is the simplest while robust way to solve the issue without performance penalty. For further improvement on customized state listener, we track in this issue: #1046 and tackle it later.